parser_test.go

```diff
@@ -1812,6 +1812,20 @@ func (s *testParserSuite) TestDDL(c *C) {
 	s.RunTest(c, table)
 }
 
+func (s *testParserSuite) TestHintError(c *C) {
+	parser := New()
+	stmt, err := parser.Parse("select /*+ tidb_unknow(T1,t2) */ c1, c2 from t1, t2 where t1.c1 = t2.c1", "", "")
+	c.Assert(err, IsNil)
+	c.Assert(len(stmt[0].(*ast.SelectStmt).TableHints), Equals, 0)
+	stmt, err = parser.Parse("select /*+ tidb_unknow(T1,t2, 1) TIDB_INLJ(t1, T2) */ c1, c2 from t1, t2 where t1.c1 = t2.c1", "", "")
+	c.Assert(len(stmt[0].(*ast.SelectStmt).TableHints), Equals, 0)
+	c.Assert(err, IsNil)
+	stmt, err = parser.Parse("select1 /*+ TIDB_INLJ(t1, T2) */ c1, c2 from t1, t2 where t1.c1 = t2.c1", "", "")
+	c.Assert(err, NotNil)
+	stmt, err = parser.Parse("select /*+ TIDB_INLJ(t1, T2) */ c1, c2 fromt t1, t2 where t1.c1 = t2.c1", "", "")
```

Comment:
fromt -> from

Reply:
those two tests just want to test whether error still can happen

Reply:
oh, i see.